### PR TITLE
Add info about the data directory to 'Compiling with mono'

### PR DESCRIPTION
**Pull request by neikeq**
Added information about the new data directory.

### CODE REVIEW

File: development/compiling/compiling_with_mono.rst [hunk 1]
```diff
@@ -102,24 +107,83 @@ Example (X11)
...
+
+- ``Api`` (optional)
+- ``Mono`` (optional)
+- ``Tools`` (required)
+
+The ``Tools`` subdirectory contains tools required by the editor, like the ``GodotSharpTools`` assembly.
+
+The ``Mono`` subdirectory is optional. It can be used to bundle the Mono framework assemblies and configuration files with the Godot editor, as well as some shared library dependencies like ``MonoPosixHelper``. This is important to avoid issues that might arise when the installed Mono version in the user's system may not be the same as the one the Godot editor was built with. You can make SCons copy these files to this subdirectory by passing the option ``copy_mono_root=yes`` when building the editor.
+
+The ``Api`` directory is also optional. Godot API assemblies are not bundled with the editor by default. Instead the Godot editor will generate and build them on the user's machine the first time they are required. This can be avoiding by generating and building them manually and placing them in this subdirectory. If the editor can find them there, it will avoid the step of generating and building them again.
```
Comment (exts) [inline development/compiling/compiling_with_mono.rst]: type near "this can be avoiding" -> avoided*

Comment (neikeq) [reply development/compiling/compiling_with_mono.rst]: Fixed, thanks!

File: development/compiling/compiling_with_mono.rst [hunk 2]
```diff
@@ -102,24 +107,83 @@ Example (X11)
     scons p=x11 target=debug tools=no module_mono_enabled=yes
     scons p=x11 target=release tools=no module_mono_enabled=yes
 
-If everything went well, apart from the normal output, SCons should have also built the *GodotSharpTools.dll* assembly and copied it together with the mono runtime shared library to the ``bin`` subdirectory.
+.. _compiling_with_mono_data_directory:
```
[the remark this answers is not in the document]
Comment (neikeq) [reply development/compiling/compiling_with_mono.rst]: Let me know if the description is clear now.

**Comment by neikeq** [comment]
Added a note about `mozroots`.

**Comment by exts** [comment]
yeah it's more clear now

**Pull request merged**